kube-kvm/provision-kube-host-kvm.sh

```diff
@@ -0,0 +1,147 @@
+  echo KUBE_VM_MEM_GIB value $KUBE_VM_MEM_GIB exceeds system resources
+  exit 1
+fi
+
+unset RUNNING_VM_IMAGE
+if virsh domstate $KUBE_VM_NAME 2>/dev/null; then
+  echo "Deleting existing domain $KUBE_VM_NAME"
+  RUNNING_VM_IMAGE=$(virsh dumpxml $KUBE_VM_NAME | grep 'source file' | grep -o "'.*'" | tr -d "'")
+  if [[ -z $RUNNING_VM_IMAGE ]]; then
+    echo "Disk for domain $KUBE_VM_NAME could not be determined. Please delete this domain manually"
+    exit 1
+  fi
+  virsh destroy "$KUBE_VM_NAME"
+  virsh undefine "$KUBE_VM_NAME"
+  rm -f "$RUNNING_VM_IMAGE"
+fi
+
+cp --reflink=auto "$KUBE_VM_IMAGE_NAME.qcow2" "$KUBE_VM_IMAGE_NAME-$KUBE_VM_NAME.qcow2"
+# TODO: Test if virbr0 is present
+virt-install \
+  --name "$KUBE_VM_NAME" \
+  --memory $(( KUBE_VM_MEM_GIB * 1024 )) \
+  --vcpus 1 \
+  --disk "$KUBE_VM_IMAGE_NAME-$KUBE_VM_NAME.qcow2" \
+  --import \
+  --network network=default \
+  --network bridge=br0 \
+  --os-variant opensuse42.2 \
+  --wait 0 \
+  --hvm
+
+echo "Sleeping 30 seconds to allow VM to start and obtain IP"
+sleep 30
+IP_WAIT_TIMEOUT=600
+unset KUBE_VM_DEFAULT_MAC
+while [[ -z "$KUBE_VM_DEFAULT_MAC" ]]; do
+  echo "Checking if default interface has MAC address"
+  KUBE_VM_DEFAULT_MAC=$(virsh domiflist "$KUBE_VM_NAME" | grep default | grep -oE '([[:xdigit:]]{2}:){5}[[:xdigit:]]{2}') || true
+  if [[ -z "$KUBE_VM_DEFAULT_MAC" ]]; then
+    if [[ $IP_WAIT_TIMEOUT -ge 0 ]]; then
+      (( IP_WAIT_TIMEOUT -= 15 ))
+      echo "MAC address not yet available. Retrying in 15 seconds"
+      sleep 15
+    else
+      echo "No MAC address for default interface on VM $KUBE_VM_NAME after 10 minutes. Check output of \`sudo virsh domiflist $KUBE_VM_NAME\`"
+      exit 1
+    fi
+  fi
+done
+unset KUBE_VM_DEFAULT_IP
+while [[ -z "$KUBE_VM_DEFAULT_IP" ]]; do
+  echo "Checking if IP lease exists for interface $KUBE_VM_DEFAULT_MAC"
+  KUBE_VM_DEFAULT_IP=$(virsh net-dhcp-leases default --mac "$KUBE_VM_DEFAULT_MAC" | grep -oE '([0-9]{1,3}\.){3}[0-9]{1,3}') || true
+  if [[ -z "$KUBE_VM_DEFAULT_IP" ]]; then
+    if [[ $IP_WAIT_TIMEOUT -ge 0 ]]; then
+      (( IP_WAIT_TIMEOUT -= 15 ))
+      echo "No IP assigned for $KUBE_VM_DEFAULT_MAC. Retrying in 15 seconds"
+      sleep 15
+    else
+      echo "No IP assigned for $KUBE_VM_DEFAULT_MAC after 10 minutes. Exiting."
+      exit 1
+    fi
+  fi
+done
+echo "Docker VM IP found: $KUBE_VM_DEFAULT_IP. Waiting for sshd to start"
+SSHD_TIMEOUT=60
+while [[ $SSHD_TIMEOUT -ge 0 ]]; do
+  nc $KUBE_VM_DEFAULT_IP 22 </dev/null | grep 'SSH-' && break
+  sleep 5
+  (( SSHD_TIMEOUT -= 5 ))
+done
+ssh-keygen -R $KUBE_VM_DEFAULT_IP
+ssh -o StrictHostKeyChecking=no -Ti "$KUBE_VM_KEY" vagrant@$KUBE_VM_DEFAULT_IP -- sudo bash -o errexit << EOF
+hostname $KUBE_VM_NAME
+echo $KUBE_VM_NAME > /etc/hostname
+cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth1
+echo DHCLIENT_SET_DEFAULT_ROUTE='yes' /etc/sysconfig/network/ifcfg-eth1
+wicked ifup eth1
```

Comment:
I… don't understand. `wicked` is, as far as I know, SUSE-specific…

Reply:
The VM should be a SUSE VM? But the host machine may not be

Reply:
Ah.. I see what you were saying about setting the hostname

Reply:
> Are we assuming SUSE here? If yes, yast dns edit hostname="${K8S_VM_NAME}" might work here.

So apparently, even without making any of my own changes, this command doesn't work for me, because both the default `/etc/sysconfig/network/ifcfg-eth0` and `/etc/sysconfig/network/dhcp` files contain the line `DHCLIENT_SET_HOSTNAME='yes'`

I think if I removed this line from both of these files and add it to ifcfg-eth1, I may be able to set the hostname with yast noninteractively, but I'm not clear on what wicked does with this option in the various config.

For simplicity, I'll stick with setting `hostname` without yast for now.